astropy/coordinates/tests/test_sites.py

```diff
@@ -86,6 +86,38 @@ def test_EarthLocation_basic():
         EarthLocation.of_site("nonexistent")
 
 
+@pytest.mark.parametrize(
+    "class_method,args",
+    [(EarthLocation.get_site_names, []), (EarthLocation.of_site, ["greenwich"])],
+)
+def test_Earthlocation_refresh_cache_is_mandatory_kwarg(args, class_method):
+    with pytest.raises(
+        TypeError,
+        match=(
+            rf".*{class_method.__name__}\(\) takes [12] positional "
+            "arguments? but [23] were given$"
+        ),
+    ):
+        class_method(*args, False)
+
+
+@pytest.mark.parametrize(
+    "class_method,args",
+    [(EarthLocation.get_site_names, []), (EarthLocation.of_site, ["greenwich"])],
+)
+@pytest.mark.parametrize("refresh_cache", [False, True])
+def test_Earthlocation_refresh_cache(args, class_method, monkeypatch, refresh_cache):
```

Comment:
```suggestion
def test_Earthlocation_refresh_cache(class_method, args, monkeypatch, refresh_cache):
```

Reply:
I will leave issues with the unit test to @eerovaher to resolve.